blockchain/merkle_test.go

```diff
@@ -21,3 +24,89 @@ func TestMerkle(t *testing.T) {
 			"got %v, want %v", calculatedMerkleRoot, wantMerkle)
 	}
 }
+
+func makeHashes(size int) []*chainhash.Hash {
+	var hashes = make([]*chainhash.Hash, size)
+	for i := range hashes {
+		hashes[i] = new(chainhash.Hash)
+	}
+	return hashes
+}
+
+func makeTxs(size int) []*btcutil.Tx {
+	var txs = make([]*btcutil.Tx, size)
+	for i := range txs {
+		tx := btcutil.NewTx(wire.NewMsgTx(2))
+		tx.Hash()
+		txs[i] = tx
+	}
+	return txs
+}
+
+// BenchmarkRollingMerkle benches the RollingMerkleTree while varying the number
+// of leaves pushed to the tree.
+func BenchmarkRollingMerkle(b *testing.B) {
+	sizes := []int{
+		1000,
+		2000,
+		4000,
+		8000,
+		16000,
+		32000,
+	}
+
+	txs := make([][]*btcutil.Tx, len(sizes))
+	for i, size := range sizes {
+		txs[i] = makeTxs(size)
+	}
+
+	for i, size := range sizes {
+		name := fmt.Sprintf("%d", size)
```

Comment:
nit: could do `txs = makeTxs(size)` here instead, and same below.

Reply:
addressed in the latest push